src/tim/CMakeLists.txt

```diff
@@ -72,16 +72,21 @@ endif()
 add_library(${TARGET_NAME} ${${TARGET_NAME}_SRCS})
 target_include_directories(${TARGET_NAME} PRIVATE ${INC_DIRS})
 target_link_libraries(${TARGET_NAME} PUBLIC
-   -Wl,--whole-archive tim_internal -Wl,--no-whole-archive ${EXTERNAL_LIBS})
+    -Wl,--no-whole-archive ${EXTERNAL_LIBS} ${OVXDRV_LIBRARIES})
 
-install(TARGETS ${TARGET_NAME} ${TARGET_NAME}
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
+if(${TIM_VX_USE_EXTERNAL_OVXLIB})
+  #-Wl,--whole-archive should not applied to external library, but only for shared library
+  target_link_libraries(${TARGET_NAME} PUBLIC tim_internal)
+endif()
 
-if (CMAKE_SYSTEM_PROCESSOR STREQUAL aarch64)
-    install(TARGETS ${TARGET_NAME} ${TARGET_NAME}
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/lib64)
+if (NOT CMAKE_INSTALL_LIBDIR)
+	set(CMAKE_INSTALL_LIBDIR "lib")
 endif()
 
+
+install(TARGETS ${TARGET_NAME} ${TARGET_NAME}
+	DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
```

Comment:
Maybe you need to specify the dst as `${CMAKE_INSTALL_PREFIX}/lib` instead of using `${CMAKE_INSTALL_LIBDIR}` to have consistent output structure. I tested on two machines, one of which produces `lib` directory under `install`, the other of which does not produce `lib` directory.